app/javascript/components/cloud-volume-form/cloud-volume-form.schema.js

```diff
@@ -0,0 +1,39 @@
+import { componentTypes, validatorTypes } from '@@ddf';
+
+const createSchema = (fields, promise, edit, loadSchema) => ({
+  fields: [
+    {
+      component: componentTypes.SELECT,
+      name: 'ext_management_system_id',
+      id: 'ext_management_system_id',
+      label: __('Storage Manager'),
+      placeholder: __('<Choose>'),
+      onChange: value => API.options(`/api/cloud_volumes?ems_id=${value}`).then(loadSchema()),
+      loadOptions: () => promise.then(({ resources }) => resources.map(({ id, name }) => ({ value: id, label: __(name) }))),
+      isDisabled: edit,
+      isRequired: true,
+      validate: [{ type: validatorTypes.REQUIRED }],
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'name',
+      id: 'name',
+      label: __('Volume Name'),
+      isRequired: true,
+      validate: [{ type: validatorTypes.REQUIRED }],
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'size',
+      id: 'size',
+      label: __('Size (in bytes)'),
+      type: 'number',
+      isRequired: true,
+      step: 1024 * 1024 * 1024,
+      validate: [{ type: validatorTypes.REQUIRED }],
+    },
+    ...fields,
```

Comment:
Some of these fields are not available when editing, but the API only can send a static schema. As a solution we could have an `edit` text field with `type: hidden` set to `true` that we omit in the `onSubmit` function:
```js
{
  component: componentTypes.TEXT_FIELD,
  name: 'edit',
  id: 'edit',
  type: 'hidden',
  value: edit,
}
```

Then I can use its value in the API as a `condition`...